api4/user_test.go

```diff
@@ -1112,6 +1115,22 @@ func TestPatchUser(t *testing.T) {
 		t.Fatal("manualTimezone did not update properly")
 	}
 
+	err := th.App.CheckPasswordAndAllCriteria(ruser, *patch.Password, "")
+	if err == nil {
+		t.Fatal("Password should not match")
+	}
+
+	currentPassword := user.Password
+	user, err = th.App.GetUser(ruser.Id)
+	if err != nil {
+		t.Fatal("User Get shouldn't error")
+	}
+
+	err = th.App.CheckPasswordAndAllCriteria(user, currentPassword, "")
+	if err != nil {
+		t.Fatal("Password should still match")
+	}
+
```

Comment:
Can we add a test case to make sure the user can't change their email without the password. Both endpoints.

Reply:
Done, see latest commit :+1: